time.go

```diff
@@ -0,0 +1,85 @@
+package weave
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/iov-one/weave/errors"
+)
+
+// UNIX time value of 0001-01-01T00:00:00Z
+const zeroUnixTime = -62135596800
```

Comment:
This is indeed the lowest value that Python supports:

```
>>> datetime.datetime.utcfromtimestamp(-62135596800)
datetime.datetime(1, 1, 1, 0, 0)
```

and much stricter than Number.MIN_SAFE_INTEGER. Makes sense to me for the general time type

Reply:
I call it zero time, because I got this value by extracting unix time from zero `time.Time` value. Go's `Time` does not seem to support anything that is before that point in time.